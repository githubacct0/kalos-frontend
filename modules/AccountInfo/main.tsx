import React, { ChangeEvent } from 'react';
import { User, UserClient } from '@kalos-core/kalos-rpc/User';
import PermIdentityIcon from '@material-ui/icons/PermIdentity';
import EditIcon from '@material-ui/icons/Edit';
import SendIcon from '@material-ui/icons/Send';
import {
  TextField,
  Grid,
  Modal,
  Button,
  Paper,
  Switch,
  Typography,
  Divider,
  IconButton,
  FormControlLabel,
  CssBaseline,
} from '@material-ui/core';
import CloseIcon from '@material-ui/icons/CloseSharp';

interface props {
  userId: number;
}

interface state {
  user: User.AsObject;
  isModalOpen: boolean;
  isEditing: boolean;
  isLoginModalOpen: boolean;
  searchString: string;
}

export class AccountInfo extends React.PureComponent<props, state> {
  UserClient: UserClient;
  oldPassword: React.RefObject<HTMLInputElement>;
  newPassword: React.RefObject<HTMLInputElement>;
  reTypePassword: React.RefObject<HTMLInputElement>;
  oldLogin: React.RefObject<HTMLInputElement>;
  newLogin: React.RefObject<HTMLInputElement>;
  reTypeLogin: React.RefObject<HTMLInputElement>;

  constructor(props: props) {
    super(props);
    this.state = {
      user: new User().toObject(),
      isModalOpen: false,
      isEditing: false,
      isLoginModalOpen: false,
      searchString: '',
    };
    this.handleUpdateLogin = this.handleUpdateLogin.bind(this);
    this.toggleLoginModal = this.toggleLoginModal.bind(this);
    this.toggleEditing = this.toggleEditing.bind(this);
    this.toggleModal = this.toggleModal.bind(this);
    this.handleUpdatePassword = this.handleUpdatePassword.bind(this);
    this.UserClient = new UserClient('https://core-dev.kalosflorida.com:8443');
    this.oldPassword = React.createRef();
    this.newPassword = React.createRef();
    this.reTypePassword = React.createRef();
    this.oldLogin = React.createRef();
    this.newLogin = React.createRef();
    this.reTypeLogin = React.createRef();
  }

  async fetchUser() {
    const req = new User();
    req.setId(this.props.userId);

    const result = await this.UserClient.Get(req);
    this.setState({
      user: result,
    });
  }

  toggleModal() {
    this.setState(prevState => ({
      isModalOpen: !prevState.isModalOpen,
    }));
  }

  toggleLoginModal() {
    this.setState(prevState => ({
      isLoginModalOpen: !prevState.isLoginModalOpen,
    }));
  }

  async handleUpdateLogin() {
    try {
      if (
        this.oldLogin.current &&
        this.newLogin.current &&
        this.reTypeLogin.current
      ) {
        const oldLogin = this.oldLogin.current.value;
        const newLogin = this.newLogin.current.value;
        const reTypeLogin = this.reTypeLogin.current.value;

        if (newLogin !== reTypeLogin) {
          throw 'Usernames do not match';
        }
        if (oldLogin !== this.state.user.login) {
          throw 'Old Login is incorrect';
        }
        const login = this.oldLogin.current.value;
        const req = new User();
        req.setLogin(newLogin);
        req.setId(this.state.user.id);
        req.setFieldMaskList(['Login']);
        const updatedUser = await this.UserClient.Update(req);
        this.setState({ user: updatedUser });

        this.toggleLoginModal();
      }
    } catch (err) {
      alert(err);
    }
  }

  async handleUpdatePassword() {
    try {
      if (
        this.oldPassword.current &&
        this.newPassword.current &&
        this.reTypePassword.current
      ) {
        const oldPassword = this.oldPassword.current.value;
        const newPassword = this.newPassword.current.value;
        const reTypePassword = this.reTypePassword.current.value;
        if (newPassword.length < 8) {
          throw 'Password needs to be 8 characters long';
        }
        if (newPassword !== reTypePassword) {
          throw 'Passwords do not match';
        }
        if (oldPassword !== this.state.user.pwd) {
          throw 'Old Password is incorrect';
        }
        const req = new User();
        req.setPwd(newPassword);
        req.setFieldMaskList(['Pwd']);
        req.setId(this.state.user.id);
        await this.UserClient.Update(req);
        this.toggleModal();
      }
    } catch (err) {
      alert(err);
    }
  }

  toggleEditing() {
    this.setState(prevState => ({
      isEditing: !prevState.isEditing,
    }));
  }

  updateUser<K extends keyof User.AsObject>(prop: K) {
    return async (e: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
      const user = new User();
      const upperCaseProp = `${prop[0].toUpperCase()}${prop.slice(1)}`;
      const methodName = `set${upperCaseProp}`;
      user.setId(this.state.user.id);
      //@ts-ignore
      user[methodName](e.target.value);
      user.setFieldMaskList([upperCaseProp]);
      const updatedUser = await this.UserClient.Update(user);
      this.setState(() => ({ user: updatedUser }));
    };
  }
  updatePassword = this.updateUser('pwd');
  updateLogin = this.updateUser('login');
  updateEmail = this.updateUser('email');
  updateCellPhone = this.updateUser('cellphone');

  updateZipCode = this.updateUser('zip');
  updateCity = this.updateUser('city');
  updateStreetAddress = this.updateUser('address');
  updateFirstName = this.updateUser('firstname');
  updateLastName = this.updateUser('lastname');

  async componentDidMount() {
    await this.UserClient.GetToken('gavinorr', 'G@vin123');
    await this.fetchUser();
  }

  render() {
    const { isEditing } = this.state;
    if (this.state.user.id === 0) {
      return null;
    }
    return (
      <>
        <CssBaseline />
        <Grid
          style={{ paddingBottom: '20px' }}
          container
          alignItems="stretch"
          justify="center"
          direction="column"
        >
          <Grid container direction="row" justify="space-evenly">
            <TextField
              disabled={!isEditing}
              style={{ marginRight: '10px' }}
              defaultValue={this.state.user.firstname}
              onChange={this.updateFirstName}
              label={'First Name'}
            />
            <TextField
              disabled={!isEditing}
              defaultValue={this.state.user.lastname}
              onChange={this.updateLastName}
              label={'Last Name'}
            />
            <FormControlLabel
              control={
                <Switch
                  checked={this.state.isEditing}
                  onChange={this.toggleEditing}
                  value="isEditing"
                  color="primary"
                />
              }
              label={
                this.state.isEditing ? 'Editing Enabled' : 'Editing Disabled'
              }
            />
          </Grid>

          <Grid
            style={{ paddingBottom: '20px', paddingTop: '20px' }}
            container
            alignItems="stretch"
            justify="flex-start"
            direction="column"
          >
            <TextField
              style={{ paddingBottom: '10px', paddingTop: '10px' }}
              disabled={!isEditing}
              defaultValue={this.state.user.address}
              onChange={this.updateStreetAddress}
              label={'Street Address'}
            />
            <TextField
              style={{ paddingBottom: '10px', paddingTop: '10px' }}
              disabled={!isEditing}
              defaultValue={this.state.user.city}
              onChange={this.updateCity}
              label={'City'}
            />
            <TextField
              style={{ paddingBottom: '10px', paddingTop: '10px' }}
              disabled={!isEditing}
              defaultValue={this.state.user.zip}
              onChange={this.updateZipCode}
              label={'Zip Code'}
            />

            <TextField
              style={{ paddingBottom: '10px', paddingTop: '10px' }}
              disabled={!isEditing}
              defaultValue={this.state.user.cellphone}
              onChange={this.updateCellPhone}
              label={'Phone Number'}
            />
            <TextField
              style={{
                paddingBottom: '10px',
                paddingTop: '10px',
                paddingRight: 10,
              }}
              disabled
              defaultValue={this.state.user.email}
              onChange={this.updateEmail}
              label={'Email'}
            />
          </Grid>
          <Grid
            container
            direction="row"
            justify="space-evenly"
            style={{ marginTop: 10 }}
          >
            <Button
              disabled={!isEditing}
              onClick={this.toggleLoginModal}
              variant="contained"
              style={{ width: '30%' }}
            >
              Change Login
            </Button>
            <Button
              disabled={!isEditing}
              onClick={this.toggleModal}
              variant="contained"
              style={{ width: '30%' }}
            >
              Change Password
            </Button>
          </Grid>
        </Grid>
        <Modal
          open={this.state.isModalOpen}
          onClose={this.toggleModal}
          style={{
            margin: '10px',
            display: 'flex',
            justifyContent: 'center',
            alignItems: 'center',
          }}
        >
          <Paper
            style={{
              padding: 20,
              width: '30%',
            }}
          >
            <Grid container direction={'column'}>
              <Grid
                container
                direction="row"
                justify="space-between"
                alignItems="center"
              >
                <Typography component="h2" variant="h6">
                  Change Password
                </Typography>
                <IconButton onClick={this.toggleModal}>
                  <CloseIcon />
                </IconButton>
              </Grid>
              <Divider />
              <TextField
                disabled={!isEditing}
                label={'Old Password'}
                type="password"
                inputProps={{ ref: this.oldPassword }}
              />
              <TextField
                disabled={!isEditing}
                label={'New Password'}
                type="password"
                inputProps={{
                  ref: this.newPassword,
                }}
              />
              <TextField
                disabled={!isEditing}
                label={'Repeat Password'}
                type="password"
                inputProps={{
                  ref: this.reTypePassword,
                }}
              />
              <Button
                endIcon={<SendIcon />}
                variant="contained"
                style={{ padding: 5, marginTop: 10 }}
                onClick={this.handleUpdatePassword}
              >
                Submit
              </Button>
            </Grid>
          </Paper>
        </Modal>
        <Modal
          open={this.state.isLoginModalOpen}
          onClose={this.toggleLoginModal}
          style={{
            margin: '10px',
            display: 'flex',
            justifyContent: 'center',
            alignItems: 'center',
          }}
        >
          <Paper
            style={{
              padding: 20,
              width: '30%',
              display: 'flex',
              flexDirection: 'column',
              justifyContent: 'center',
              alignItems: 'center',
            }}
          >
            <Grid
              container
              direction="row"
              justify="space-between"
              alignItems="center"
            >
              <Typography component="h1" variant="h6">
                Change Login
              </Typography>
              <IconButton onClick={this.toggleLoginModal}>
                <CloseIcon />
              </IconButton>
            </Grid>
            <Divider />
            <TextField
              fullWidth
              defaultValue={this.state}
              disabled={!isEditing}
              label={'Old Login'}
              inputProps={{
                ref: this.oldLogin,
              }}
            />
            <TextField
              fullWidth
              disabled={!isEditing}
              label={'New Login'}
              inputProps={{
                ref: this.newLogin,
              }}
            />
            <TextField
              fullWidth
              disabled={!isEditing}
              label={' ReType Login'}
              inputProps={{
                ref: this.reTypeLogin,
              }}
            />
            <Button
              endIcon={<SendIcon />}
              variant="contained"
              style={{ padding: 5, marginTop: 10, width: '100%' }}
              onClick={this.handleUpdateLogin}
            >
              Confirm
            </Button>
          </Paper>
        </Modal>
      </>
    );
  }
}
