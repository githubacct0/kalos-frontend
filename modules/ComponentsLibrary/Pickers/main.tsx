// this files ts-ignore lines have been checked
import React from 'react';
import InputLabel from '@material-ui/core/InputLabel';
import FormHelperText from '@material-ui/core/FormHelperText';
import FormControl from '@material-ui/core/FormControl';
import NativeSelect from '@material-ui/core/NativeSelect';
import { BaseClient } from '@kalos-core/kalos-rpc/BaseClient';

interface props<T, C> {
  selected: number;
  onSelect?(acc: T): void;
  client: C;
  id: string;
  reqObj: T;
}

interface state<T> {
  list: T[];
}

export class Picker<
  T extends { id: number; description: string },
  C extends BaseClient,
> extends React.PureComponent<props<T, C>, state<T>> {
  Client: C;
  constructor(props: props<T, C>) {
    super(props);
    this.state = {
      list: [],
    };
    //@ts-ignore
    this.Client = new this.props.client();

    this.handleSelect = this.handleSelect.bind(this);
    this.addItem = this.addItem.bind(this);
  }

  handleSelect(e: React.SyntheticEvent<HTMLSelectElement>) {
    const id = parseInt(e.currentTarget.value);
    if (this.props.onSelect) {
      const acc = this.state.list.find(a => a.id === id);
      if (acc) {
        try {
          this.props.onSelect(acc);
        } catch (err) {
          console.log(err);
        }
      }
    }
  }

  addItem(item: T) {
    this.setState(prevState => ({
      list: prevState.list.concat(item),
    }));
  }

  async fetchItems() {
    //@ts-ignore
    this.Client.List(new T(), this.addItem);
  }

  componentDidMount() {
    this.fetchItems();
  }

  render() {
    return (
      <FormControl
        style={{
          margin: 5,
          minWidth: 120,
        }}
      >
        <InputLabel htmlFor="cost-center-picker">Cost Center</InputLabel>
        <NativeSelect
          value={this.props.selected}
          onChange={this.handleSelect}
          inputProps={{ id: 'cost-center-picker' }}
        >
          <option value={0}>Select Cost Center</option>
          {this.state.list.map(item => (
            <option value={item.id} key={`${item.description}-${item.id}`}>
              {item.description}
            </option>
          ))}
        </NativeSelect>
        {/*<FormHelperText>
          Assign a purchase category to your receipt
        </FormHelperText>*/}
      </FormControl>
    );
  }
}
