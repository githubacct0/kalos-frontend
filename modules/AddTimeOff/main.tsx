import React, { FC } from 'react';
import { PageWrapper, PageWrapperProps } from '../PageWrapper/main';
import { TimeOff, Props as TimeOffProps } from '../ComponentsLibrary/TimeOff';

export const AddTimeOff: FC<TimeOffProps & PageWrapperProps> = props => (
  <PageWrapper {...props} userID={props.loggedUserId} withHeader>
    <TimeOff {...props} />
  </PageWrapper>
);
