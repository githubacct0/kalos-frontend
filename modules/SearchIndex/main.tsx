import React, { FC } from 'react';
import { AdvancedSearch } from '../ComponentsLibrary/AdvancedSearch';
import { PageWrapper } from '../PageWrapper/main';

interface Props {
  loggedUserId: number;
}

export const SearchIndex: FC<Props> = ({ loggedUserId }) => (
  <PageWrapper userID={loggedUserId}>
    <AdvancedSearch
      loggedUserId={loggedUserId}
      title="Search"
      kinds={['serviceCalls', 'customers', 'properties']}
      editableCustomers
      editableProperties
    />
  </PageWrapper>
);
