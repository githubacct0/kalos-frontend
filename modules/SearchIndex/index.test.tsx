import React from 'react';
import renderer from 'react-test-renderer';
import { SearchIndex } from './main';

test('SearchIndex renders correctly', () => {
  const tree = renderer
    .create(<SearchIndex loggedUserId={101253} withHeader />)
    .toJSON();
  expect(tree).toMatchSnapshot(); // Rendered correctly
});
