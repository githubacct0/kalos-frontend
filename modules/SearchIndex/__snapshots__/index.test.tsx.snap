// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchIndex renders correctly 1`] = `
<div
  className="PageWrapper"
>
  <div
    className="PageWrapperMenu"
  />
  <div
    className="PageWrapperContent"
    style={
      Object {
        "padding": "0rem",
      }
    }
  >
    <div>
      <div
        className="SectionBar fixedActions sticky"
      >
        <div
          className="SectionBarHeaderWrapper"
        >
          <div
            className="SectionBarHeader"
          >
            <div
              className="SectionBarTitleWrapper"
            >
              <h5
                className="MuiTypography-root SectionBarTitle MuiTypography-h5"
                onClick={[Function]}
              >
                Search
                 
              </h5>
            </div>
          </div>
          <div
            className="SectionBarActions"
          />
        </div>
      </div>
      <form
        className="AdvancedSearchForm PlainForm compact"
        onSubmit={[Function]}
      >
        <div
          className="PlainFormGroup"
        >
          <div
            className="Field"
            style={Object {}}
          >
            <div
              className="MuiFormControl-root FieldInput MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
                data-shrink={true}
                id="kind-select-label"
              >
                Search
                
              </label>
              <div
                className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                onClick={[Function]}
              >
                <div
                  aria-haspopup="listbox"
                  aria-labelledby="kind-select-label kind-select"
                  className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                  id="kind-select"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onMouseDown={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  Customers
                </div>
                <input
                  aria-hidden={true}
                  className="MuiSelect-nativeInput"
                  onAnimationStart={[Function]}
                  onChange={[Function]}
                  required={false}
                  tabIndex={-1}
                  value="customers"
                />
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSelect-icon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </div>
               
            </div>
          </div>
          <div
            className="Field PlainFormField type-search"
            style={Object {}}
          >
            <div
              className="MuiFormControl-root MuiTextField-root FieldInput MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
                data-shrink={true}
              >
                First Name
                
              </label>
              <div
                className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiInput-input MuiInputBase-inputTypeSearch MuiInput-inputTypeSearch"
                  disabled={false}
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  readOnly={false}
                  required={false}
                  type="search"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="Field PlainFormField type-search"
            style={Object {}}
          >
            <div
              className="MuiFormControl-root MuiTextField-root FieldInput MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
                data-shrink={true}
              >
                Last Name
                
              </label>
              <div
                className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiInput-input MuiInputBase-inputTypeSearch MuiInput-inputTypeSearch"
                  disabled={false}
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  readOnly={false}
                  required={false}
                  type="search"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="Field PlainFormField type-search"
            style={Object {}}
          >
            <div
              className="MuiFormControl-root MuiTextField-root FieldInput MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
                data-shrink={true}
              >
                Business Name
                
              </label>
              <div
                className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiInput-input MuiInputBase-inputTypeSearch MuiInput-inputTypeSearch"
                  disabled={false}
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  readOnly={false}
                  required={false}
                  type="search"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="PlainFormGroup"
        >
          <div
            className="Field type-search"
            style={Object {}}
          >
            <div
              className="MuiFormControl-root MuiTextField-root FieldInput MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
                data-shrink={true}
              >
                Email
                
              </label>
              <div
                className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiInput-input MuiInputBase-inputTypeSearch MuiInput-inputTypeSearch"
                  disabled={false}
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  readOnly={false}
                  required={false}
                  type="search"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="Field PlainFormField type-search"
            style={Object {}}
          >
            <div
              actions={
                Array [
                  Object {
                    "label": "Reset",
                    "onClick": [Function],
                    "variant": "outlined",
                  },
                  Object {
                    "label": "Search",
                    "onClick": [Function],
                  },
                ]
              }
              className="MuiFormControl-root MuiTextField-root FieldInput MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
                data-shrink={true}
              >
                Primary Phone
                
              </label>
              <div
                className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiInput-input MuiInputBase-inputTypeSearch MuiInput-inputTypeSearch"
                  disabled={false}
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  readOnly={false}
                  required={false}
                  type="search"
                  value=""
                />
              </div>
            </div>
            <div
              className="Actions FieldActions"
            >
              <div
                className="ActionsActions responsiveColumn"
              >
                <button
                  className="MuiButtonBase-root MuiButton-root MuiButton-outlined ButtonWrapper ActionsButton responsiveColumn size-small status-undefined variant-outlined MuiButton-outlinedPrimary MuiButton-outlinedSizeSmall MuiButton-sizeSmall"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  <span
                    className="MuiButton-label"
                  >
                    Reset
                  </span>
                </button>
                <button
                  className="MuiButtonBase-root MuiButton-root MuiButton-contained ButtonWrapper ActionsButton responsiveColumn size-small status-undefined variant-contained MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  <span
                    className="MuiButton-label"
                  >
                    Search
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
        <button
          className="PlainFormSubmit"
          type="submit"
        >
          Submit
        </button>
      </form>
      <div
        className="InfoTable"
      >
        <div
          className="InfoTableHeader"
        >
          <div
            className="MuiTypography-root InfoTableColumn MuiTypography-body1"
            style={
              Object {
                "flexGrow": 0,
                "flexShrink": 1,
                "width": "16.666666666666668%",
              }
            }
          >
            <span
              className="InfoTableDir"
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                  "justifyContent": "flex-start",
                }
              }
            >
              First Name
               
            </span>
          </div>
          <div
            className="MuiTypography-root InfoTableColumn MuiTypography-body1"
            style={
              Object {
                "flexGrow": 0,
                "flexShrink": 1,
                "width": "16.666666666666668%",
              }
            }
          >
            <span
              className="InfoTableDir"
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                  "justifyContent": "flex-start",
                }
              }
            >
              Last Name
               
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 14l5-5 5 5z"
                />
              </svg>
            </span>
          </div>
          <div
            className="MuiTypography-root InfoTableColumn MuiTypography-body1"
            style={
              Object {
                "flexGrow": 0,
                "flexShrink": 1,
                "width": "16.666666666666668%",
              }
            }
          >
            <span
              className="InfoTableDir"
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                  "justifyContent": "flex-start",
                }
              }
            >
              Business Name
               
            </span>
          </div>
          <div
            className="MuiTypography-root InfoTableColumn MuiTypography-body1"
            style={
              Object {
                "flexGrow": 0,
                "flexShrink": 1,
                "width": "16.666666666666668%",
              }
            }
          >
            <span
              className="InfoTableDir"
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                  "justifyContent": "flex-start",
                }
              }
            >
              Primary Phone
               
            </span>
          </div>
          <div
            className="MuiTypography-root InfoTableColumn MuiTypography-body1"
            style={
              Object {
                "flexGrow": 0,
                "flexShrink": 1,
                "width": "16.666666666666668%",
              }
            }
          >
            <span
              className="InfoTableDir"
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                  "justifyContent": "flex-start",
                }
              }
            >
              Email
               
            </span>
          </div>
          <div
            className="MuiTypography-root InfoTableColumn MuiTypography-body1"
            style={
              Object {
                "flexGrow": 0,
                "flexShrink": 1,
                "width": "16.666666666666668%",
              }
            }
          >
            <span
              className="InfoTableDir"
              style={
                Object {
                  "cursor": "default",
                  "justifyContent": "flex-start",
                }
              }
            >
              
               
            </span>
          </div>
        </div>
        <div
          className="InfoTableRow hoverable"
        >
          <div
            className="MuiTypography-root InfoTableItem MuiTypography-body1"
            style={
              Object {
                "cursor": "default",
                "flexGrow": 0,
                "flexShrink": 1,
                "justifyContent": "flex-start",
                "width": "16.666666666666668%",
              }
            }
          >
            <span
              className="InfoTableFake"
            />
          </div>
          <div
            className="MuiTypography-root InfoTableItem MuiTypography-body1"
            style={
              Object {
                "cursor": "default",
                "flexGrow": 0,
                "flexShrink": 1,
                "justifyContent": "flex-start",
                "width": "16.666666666666668%",
              }
            }
          >
            <span
              className="InfoTableFake"
            />
          </div>
          <div
            className="MuiTypography-root InfoTableItem MuiTypography-body1"
            style={
              Object {
                "cursor": "default",
                "flexGrow": 0,
                "flexShrink": 1,
                "justifyContent": "flex-start",
                "width": "16.666666666666668%",
              }
            }
          >
            <span
              className="InfoTableFake"
            />
          </div>
          <div
            className="MuiTypography-root InfoTableItem MuiTypography-body1"
            style={
              Object {
                "cursor": "default",
                "flexGrow": 0,
                "flexShrink": 1,
                "justifyContent": "flex-start",
                "width": "16.666666666666668%",
              }
            }
          >
            <span
              className="InfoTableFake"
            />
          </div>
          <div
            className="MuiTypography-root InfoTableItem MuiTypography-body1"
            style={
              Object {
                "cursor": "default",
                "flexGrow": 0,
                "flexShrink": 1,
                "justifyContent": "flex-start",
                "width": "16.666666666666668%",
              }
            }
          >
            <span
              className="InfoTableFake"
            />
          </div>
          <div
            className="MuiTypography-root InfoTableItem MuiTypography-body1"
            style={
              Object {
                "cursor": "default",
                "flexGrow": 0,
                "flexShrink": 1,
                "justifyContent": "flex-start",
                "width": "16.666666666666668%",
              }
            }
          >
            <span
              className="InfoTableFake"
            />
          </div>
        </div>
        <div
          className="InfoTableRow hoverable"
        >
          <div
            className="MuiTypography-root InfoTableItem MuiTypography-body1"
            style={
              Object {
                "cursor": "default",
                "flexGrow": 0,
                "flexShrink": 1,
                "justifyContent": "flex-start",
                "width": "16.666666666666668%",
              }
            }
          >
            <span
              className="InfoTableFake"
            />
          </div>
          <div
            className="MuiTypography-root InfoTableItem MuiTypography-body1"
            style={
              Object {
                "cursor": "default",
                "flexGrow": 0,
                "flexShrink": 1,
                "justifyContent": "flex-start",
                "width": "16.666666666666668%",
              }
            }
          >
            <span
              className="InfoTableFake"
            />
          </div>
          <div
            className="MuiTypography-root InfoTableItem MuiTypography-body1"
            style={
              Object {
                "cursor": "default",
                "flexGrow": 0,
                "flexShrink": 1,
                "justifyContent": "flex-start",
                "width": "16.666666666666668%",
              }
            }
          >
            <span
              className="InfoTableFake"
            />
          </div>
          <div
            className="MuiTypography-root InfoTableItem MuiTypography-body1"
            style={
              Object {
                "cursor": "default",
                "flexGrow": 0,
                "flexShrink": 1,
                "justifyContent": "flex-start",
                "width": "16.666666666666668%",
              }
            }
          >
            <span
              className="InfoTableFake"
            />
          </div>
          <div
            className="MuiTypography-root InfoTableItem MuiTypography-body1"
            style={
              Object {
                "cursor": "default",
                "flexGrow": 0,
                "flexShrink": 1,
                "justifyContent": "flex-start",
                "width": "16.666666666666668%",
              }
            }
          >
            <span
              className="InfoTableFake"
            />
          </div>
          <div
            className="MuiTypography-root InfoTableItem MuiTypography-body1"
            style={
              Object {
                "cursor": "default",
                "flexGrow": 0,
                "flexShrink": 1,
                "justifyContent": "flex-start",
                "width": "16.666666666666668%",
              }
            }
          >
            <span
              className="InfoTableFake"
            />
          </div>
        </div>
        <div
          className="InfoTableRow hoverable"
        >
          <div
            className="MuiTypography-root InfoTableItem MuiTypography-body1"
            style={
              Object {
                "cursor": "default",
                "flexGrow": 0,
                "flexShrink": 1,
                "justifyContent": "flex-start",
                "width": "16.666666666666668%",
              }
            }
          >
            <span
              className="InfoTableFake"
            />
          </div>
          <div
            className="MuiTypography-root InfoTableItem MuiTypography-body1"
            style={
              Object {
                "cursor": "default",
                "flexGrow": 0,
                "flexShrink": 1,
                "justifyContent": "flex-start",
                "width": "16.666666666666668%",
              }
            }
          >
            <span
              className="InfoTableFake"
            />
          </div>
          <div
            className="MuiTypography-root InfoTableItem MuiTypography-body1"
            style={
              Object {
                "cursor": "default",
                "flexGrow": 0,
                "flexShrink": 1,
                "justifyContent": "flex-start",
                "width": "16.666666666666668%",
              }
            }
          >
            <span
              className="InfoTableFake"
            />
          </div>
          <div
            className="MuiTypography-root InfoTableItem MuiTypography-body1"
            style={
              Object {
                "cursor": "default",
                "flexGrow": 0,
                "flexShrink": 1,
                "justifyContent": "flex-start",
                "width": "16.666666666666668%",
              }
            }
          >
            <span
              className="InfoTableFake"
            />
          </div>
          <div
            className="MuiTypography-root InfoTableItem MuiTypography-body1"
            style={
              Object {
                "cursor": "default",
                "flexGrow": 0,
                "flexShrink": 1,
                "justifyContent": "flex-start",
                "width": "16.666666666666668%",
              }
            }
          >
            <span
              className="InfoTableFake"
            />
          </div>
          <div
            className="MuiTypography-root InfoTableItem MuiTypography-body1"
            style={
              Object {
                "cursor": "default",
                "flexGrow": 0,
                "flexShrink": 1,
                "justifyContent": "flex-start",
                "width": "16.666666666666668%",
              }
            }
          >
            <span
              className="InfoTableFake"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
